paddle/framework/CMakeLists.txt

```diff
@@ -84,3 +84,9 @@ cc_test(op_kernel_type_test SRCS op_kernel_type_test.cc DEPS place device_contex
 cc_test(cow_ptr_tests SRCS details/cow_ptr_test.cc)
 nv_test(data_device_transform_test SRCS data_device_transform_test.cu
         DEPS operator op_registry init math_function)
+      
+if(NOT WITH_C_API AND WITH_FLUID)
```

Comment:
Do we have to call CMake's install function in these CMakeLists.txt files? 

I am afraid that this would break the Bazel-style of the CMakeLists.txt files, which, was achieved by inventing the generic.cmake file.

Could we not install these header and library files, but leave them in the source directory and the build directory, but let users just build PaddlePaddle and include/link to these header files and library files?

Reply:
> Do we have to call CMake's install function in these CMakeLists.txt files?

Yes, We need to install `*.h` files for inference package.

>  I am afraid that this would break the Bazel-style of the CMakeLists.txt files, which, was achieved by inventing the generic.cmake file.

The style of CMake's install function is indeed different from Bazel-style. However, Bazel doesn't have a similar install function. Thus, it's hard for the user to deploy it. For example, [building-tensorflow-as-a-standalone-project](http://tuatini.me/building-tensorflow-as-a-standalone-project/) need to manually copy include and libraries for users.
![image](https://user-images.githubusercontent.com/6836917/35032298-5e7bc6e2-fba1-11e7-87dd-818b2dd0b510.png)

> Could we not install these header and library files, but leave them in the source directory and the build directory, but let users just build PaddlePaddle and include/link to these header files and library files?

I think that `make install` is necessary and will make the deploy much easier. 
- Users can deploy it without building PaddlePaddle. For example like C-API, users can directly download the paddle.tgz in [here](http://www.paddlepaddle.org/docs/develop/documentation/zh/getstarted/build_and_install/pip_install_cn.html#id15). 
- For deploy in mobile, if we don't `make install`, users should manually copy corresponding header files and library files to their project like [building-tensorflow-as-a-standalone-project](http://tuatini.me/building-tensorflow-as-a-standalone-project/).
- We will continually build for minimum library size and header files, without `make install` automatically, users may not get the updated inference package at the first time.